hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/clean/CleanPlanActionExecutor.java

```diff
@@ -192,6 +198,19 @@ protected Option<HoodieCleanerPlan> requestClean(String startCleanTime) {
     return option;
   }
 
+  private void validateForLatestTimestamp(HoodieInstant cleanInstant) {
+    try {
+      if (!skipLocking) {
+        txnManager.beginTransaction(Option.of(cleanInstant), Option.empty());
+      }
+      table.validateForLatestTimestamp(cleanInstant.getTimestamp());
+    } finally {
+      if (!skipLocking) {
```

Comment:
is the lock only used for timestamp validation? so we still support concurrent cleaning?

Reply:
timestamp validation is just to ensure the timestamp chosen for this cleaning is higher than all other timestamps generated so far (based in the timeline). 
So, does not mean concurrent cleaning. 

We could have had a concurrent compaction instant which got added to timeline just around the same time clean instant generation happened, but has higher timestamp compared to this clean instant. Just that the clean planning took non trivial amount of time during which, the compaction plan was added to the timeline